src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/time.html.twig

```diff
@@ -142,422 +138,78 @@
+<style type="text/css">
+{% autoescape 'css' %}
+    {% include '@WebProfiler/Collector/time.css.twig' %}
+{% endautoescape %}
+</style>
+<script>
+{% autoescape 'js' %}
+    {% include '@WebProfiler/Collector/time.js.twig' %}
+{% endautoescape %}
+</script>
 {% endblock %}
 
 {% macro dump_request_data(token, events, origin) %}
 {% autoescape 'js' %}
 {% from _self import dump_events %}
-                {
-                    "id": "{{ token }}",
-                    "left": {{ "%F"|format(events.__section__.origin - origin) }},
-                    "events": [
-{{ dump_events(events) }}
-                    ]
-                }
+{
+    id: "{{ token }}",
+    left: {{ "%F"|format(events.__section__.origin - origin) }},
+    end: "{{ '%F'|format(events.__section__.endtime) }}",
+    events: [ {{ dump_events(events) }} ],
+},
 {% endautoescape %}
 {% endmacro %}
 
 {% macro dump_events(events) %}
 {% autoescape 'js' %}
 {% for name, event in events %}
 {% if '__section__' != name %}
-                        {
-                            "name": "{{ name }}",
-                            "category": "{{ event.category }}",
-                            "origin": {{ "%F"|format(event.origin) }},
-                            "starttime": {{ "%F"|format(event.starttime) }},
-                            "endtime": {{ "%F"|format(event.endtime) }},
-                            "duration": {{ "%F"|format(event.duration) }},
-                            "memory": {{ "%.1F"|format(event.memory / 1024 / 1024) }},
-                            "periods": [
-                                {%- for period in event.periods -%}
-                                    {"start": {{ "%F"|format(period.starttime) }}, "end": {{ "%F"|format(period.endtime) }}}{{ loop.last ? '' : ', ' }}
-                                {%- endfor -%}
-                            ]
-                        }{{ loop.last ? '' : ',' }}
+{
+    name: "{{ name }}",
+    category: "{{ event.category }}",
+    origin: {{ "%F"|format(event.origin) }},
+    starttime: {{ "%F"|format(event.starttime) }},
+    endtime: {{ "%F"|format(event.endtime) }},
+    duration: {{ "%F"|format(event.duration) }},
+    memory: {{ "%.1F"|format(event.memory / 1024 / 1024) }},
+    periods: [
+        {%- for period in event.periods -%}
+        {
+            start: {{ "%F"|format(period.starttime) }},
+            end: {{ "%F"|format(period.endtime) }}
+        },
+        {%- endfor -%}
+    ],
+},
 {% endif %}
 {% endfor %}
 {% endautoescape %}
 {% endmacro %}
 
-{% macro display_timeline(id, colors) %}
+{% macro display_timeline(token, classnames, events, origin) %}
+{% import _self as helper %}
     <div class="sf-profiler-timeline">
         <div class="legends">
-            {% for category, color in colors %}
-                <span data-color="{{ color }}">{{ category }}</span>
+            {% for category, classname in classnames %}
+                <span class="{{classname}}">{{ category }}</span>
```

Comment:
please keep the spaces inside the ``{{ }}``, to match the official Twig coding standards.